tests/kv679_tombstone2.erl

```diff
@@ -0,0 +1,165 @@
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+%%% @copyright (C) 2014, Basho Technologies
+%%% @doc
+%%% riak_test for kv679 another doomstone flavour.
+%%%
+%%% issue kv679 is a possible dataloss issue, it's basically caused by
+%%% the fact that per key logical clocks can go backwards in time in
+%%% certain situations. The situation under test here is as follows:
+%%% Create value. Delete value (write tombstone reap tombstone from
+%%% all but one crashed primary). Write new value. Crashed primary
+%%% comes back, read repair tombstone dominates the new value. This
+%%% test depends on things happening inside a certain time limit, so
+%%% technically it is not determenistic. If you think of a better way,
+%%% please let me know.
+%%% @end
+
+-module(kv679_tombstone2).
+-behavior(riak_test).
+-compile([export_all]).
+-export([confirm/0]).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-define(BUCKET, <<"kv679">>).
+-define(KEY, <<"test">>).
+
+confirm() ->
+    Config = [{riak_kv, [{delete_mode, 10000}]}, %% 20 seconds to reap.
+              {riak_core, [{ring_creation_size, 8},
+                           {vnode_management_timer, 1000},
+                           {handoff_concurrency, 100},
+                           {vnode_inactivity_timeout, 1000}]}],
+
+    %% 4 'cos I want a perfect preflist
+    Nodes = rt:build_cluster(4, Config),
+
+    Clients =  kv679_tombstone:create_pb_clients(Nodes),
+
+    %% Get preflist for key
+    PL = kv679_tombstone:get_preflist(hd(Nodes)),
+
+    ?assert(perfect_preflist(PL)),
+
+    %% Patsy is the node that will take a fall
+    {CoordClient, Patsy} = get_coord_client_and_patsy(Clients, PL),
+
+    lager:info("CoordClient ~p~nPatsy ~p~n", [CoordClient, Patsy]),
+
+    %% Write key some times
+    kv679_tombstone:write_key(CoordClient, [<<"bob">>, <<"phil">>, <<"pete">>]),
+
+    dump_clock(CoordClient),
+
+    lager:info("wrote key thrice"),
+
+    delete_key(CoordClient),
+
+    lager:info("deleted key"),
+
+    %% kill the patsy, must happen before the reap
+    %%
+    %% @TODO make deterministic (would be nice without all these
+    %% timers)
+    timer:sleep(500),
+
+    rt:brutal_kill(Patsy),
+
+    lager:info("killed the patsy"),
+
+    %% A time to reap
+    %% %% wait for the up  nodes to reap
+    timer:sleep(15000),
+
+    lager:info("tombstone (should be) reaped"),
+
+    %% %% write the key again, this will start a new clock, a clock
+    %% that is in the past of that un-reaped primary tombstone. We use the
+    %% same node to get the same clock.
+    kv679_tombstone:write_key(CoordClient, [<<"jon">>],
+                              [{pw, 2},
+                               {sloppy_quorum, false},
+                               {n_val, 2}]),
```

Comment:
There's a reason, and I really wish I'd made a note of it in the comments. Probably to do with fallbacks, but let me try without all this and see.